### PR TITLE
Transaction gateway library

### PR DESCRIPTION
**Pull request by josearmandohf**
Add the base code of transaction gateway library.

### CODE REVIEW

**Comment by brondavies** [comment]
We'll merge this once Tim is finished with his fixes to the sync process.  Some merging of those changes into this library may need to take place first.

**Comment by aternex** [comment]
You might consider merging master back into this branch to reduce the calculated differences shown in the files changed tab. I believe a lot of these changes are actually due to being present in master and not this branch. It wouldn't actually affect the end result, but might make the PR itself easier to navigate. Bonus points for merge --squash 🥇.

File: TransactionGateway/Gateways/PushPay/PushPayPayment.cs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+            return await _pushpay.GetPayment(merchants.FirstOrDefault().Key, paymentToken);            
+        }
+
+        public static string OneTimeRedirect(string PushpayGivingBaseUrl, string Merchant, Person person, CmsData.Organization organization)
+        {
+            string givingLink = String.Format("{0}/{1}", PushpayGivingBaseUrl, Merchant);
+            string sr = String.Format("Org_{0}", organization.OrganizationId);
+            string ru = Merchant;
+            string rcv = "false";
+            string redirectUrl = givingLink
```
Comment (brondavies) [inline TransactionGateway/Gateways/PushPay/PushPayPayment.cs]: use string interpolation here instead

File: TransactionGateway/Gateways/PushPay/PushPayPayment.cs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+            return String.Format("{0}/{1}rcv=false", PushpayGivingBaseUrl, Merchant);                        
+        }
+
+        public static string RecurringGivingRedirect(string PushpayAPIBaseUrl, string Merchant)
+        {
+            string givingLink = String.Format("{0}/{1}", PushpayAPIBaseUrl, Merchant);
+            string sr = String.Format("Org_{0}", "1");
+            string ru = Merchant;
+            string rcv = "false";
+            string redirectUrl = givingLink
```
Comment (brondavies) [inline TransactionGateway/Gateways/PushPay/PushPayPayment.cs]: and here.  Also looks like there's opportunity to de-duplicate code.

**Comment by brondavies** [comment]
I just had a couple of comments, but let's make sure the batch service gets updated to use this library as soon as possible.

File: TransactionGateway/Gateways/PushPay/PushPayPayment.cs (hunk 2 above)
Comment (brondavies) [reply TransactionGateway/Gateways/PushPay/PushPayPayment.cs]: Take a look at https://www.codefactor.io/repository/github/bvcms/bvcms/pull/379 for some duplicated code blocks that should be merged/migrated.

**Pull request closed without merging**